pxemgr/pxemanager.go

```diff
@@ -340,13 +372,17 @@ func (mgr *pxeManagerT) getNextInternalIP() net.IP {
 	return net.IP{}
 }
 
-func (mgr *pxeManagerT) thisHost() string {
+func (mgr *pxeManagerT) apiURL() string {
 	scheme := "https"
 	if mgr.noTLS {
 		scheme = "http"
 	}
 
-	return fmt.Sprintf("%s://%s:%d", scheme, mgr.config.Network.BindAddr, mgr.httpPort)
+	return fmt.Sprintf("%s://%s:%d", scheme, mgr.config.Network.BindAddr, mgr.apiPort)
```

Comment:
I think it would be more safe and convenient to use https://golang.org/pkg/net/url/#URL. It implements the stringer interface so this should work quite well.